rct229/rules/section6/section6rule2.py

```diff
@@ -20,11 +22,8 @@ def __init__(self):
             each_rule=Section6Rule2.SpaceRule(),
             index_rmr="proposed",
             id="6-2",
-            description="Spaces in proposed building with hardwired lighting, including Hotel/Motel Guest Rooms, "
-            "Dormitory Living Quarters, Interior Lighting Power >= Table 9.6.1 or the lighting design in "
-            "the user model, whichever is greater; For Dwelling Units, Interior Lighting Power >= "
-            "0.6W/sq.ft. or the lighting design in the user model, whichever is greater",
-            list_path="ruleset_model_instances[0].buildings[*].zones[*].spaces[*]",
+            description="SSpaces in proposed building with hardwired lighting, including Hotel/Motel Guest Rooms, Dormitory Living Quarters, Interior Lighting Power >= Table 9.6.1; For Dwelling Units, Interior Lighting Power >= 0.6W/sq.ft.",
+            list_path="ruleset_model_instances[0].buildings[*].building_segments[*].zones[*].spaces[*]",
         )
 
```

Comment:
@weilixu   Do we want to use
```
def list_filter(self, , context_item, data):
    lighting_space_type_p = context_item.proposed["lighting_space_type"]
    return lighting_space_type_p in [
        GUEST_ROOM,
        DORMITORY_LIVING_QUARTERS
        DWELLING_UNIT
    ]
```
instead of using an _is_applicable_ method in SpaceRule?
I honestly don't know which is better here.
If we use _list_filter_, only the spaces with these lighting space types will be in the outcome.
If we use an _is_applicable_ method in SpaceRule, then all the other spaces will be included in the outcome and be set to not applicable.

My own personal preference would be to keep the outcome shorter by using the list_filter.

Reply:
Following the RDS, I think we should use `is_applicable` method in SpaceRule since this is part of the applicability check in the RDS.

However, I agree that there could be too many `not_applicable` cases that could make the export report look bad.

I think we shall be consistent, I like the `list_filter` better. In that case, we need to update the RDS to remove the applicability check.

BTW, the `list_filter` shall be under the `Section6Rule2` class. It is misplaced under the `SpaceRule` class.
@yunjoonjung-PNNL

Reply:
@weilixu This is a fairly big question.  We haven't been very consistent about this so far.  Logically,  _list_filter_ and _is_applicable_ are interchangeable.  Going forward, on what basis do we decide which to use for a particular rule?

So far, we have tended to use list_filter when the applicability test _feels_ like filtering.  I guess "feels like" here means that the filter code is simple. Another determiner seems to have been how explicit the RDS is about the applicability check.  But I suspect they write the RDS in a way that seems natural and are not necessarily trying to indicate how they want the outcome to look in this respect.

Going forward, I'd like for us to have a clear guideline for which approach to take so we don't second guess each other in the review process and add extra work.

Reply:
> @weilixu This is a fairly big question. We haven't been very consistent about this so far. Logically, _list_filter_ and _is_applicable_ are interchangeable. Going forward, on what basis do we decide which to use for a particular rule?
> 
> So far, we have tended to use list_filter when the applicability test _feels_ like filtering. I guess "feels like" here means that the filter code is simple. Another determiner seems to have been how explicit the RDS is about the applicability check. But I suspect they write the RDS in a way that seems natural and are not necessarily trying to indicate how they want the outcome to look in this respect.
> 
> Going forward, I'd like for us to have a clear guideline for which approach to take so we don't second guess each other in the review process and add extra work.

I agree.
Generally, I believe the way to use `is_applicable` is when RDS explicitly added the logic of applicability check, and use `list_filer` when RDS use ` if xxx == xxx`, example -> `If surface.type == Floor`.
This has been consistent until this RDS where the same condition appears in both applicability check and `list_filter`. 
One thing we can do is to clarify it with the RDS team to give a clear instruction on which one to choose in this particular RDS?

Reply:
Confirm with the [issue ](https://github.com/pnnl/ruleset-checking-tool/issues/605) here,
For any rule that checks a subset of data group:
1. loop through the full set of data group
2. Skip the data group that do not match the selection criteria
3. At rule assertion level (post-process), add logic that no results returned, then set the rule outcome to `NOT_APPLICABLE`